test/integration_tests/src/test_control_connection.cpp

```diff
@@ -465,4 +465,43 @@ BOOST_AUTO_TEST_CASE(randomized_contact_points)
   BOOST_CHECK_EQUAL(starting_host, get_executing_host(session));
 }
 
+/**
+ * Control connection is properly closed when invalid DC is used in LBP.
+ *
+ * This test will create a session using a DC aware LBP where the assigned DC is
+ * invalid (e.g. does not exist) and ensure the driver does not "hang" and
+ * terminates the control connection.
+ *
+ * @since 2.6.0
+ * @test_category control_connection
+ * @jira_ticket CPP-398
+ * @expected_result Driver will not hang and session/control connection will
+ *                  terminate; CASS_ERROR_LIB_NO_HOSTS_AVAILABLE
+ */
+BOOST_AUTO_TEST_CASE(invalid_dc)
+{
+  // Create the CCM cluster if it does not already exist (2 DCs)
```

Comment:
Do you need 2 DCs for this test? It seems to me you just need the one (which will be considered remote since "invalid_dc" is considered local).